core/src/main/scala/kafka/server/Server.scala

```diff
@@ -91,4 +95,12 @@ object Server {
     reporters
   }
 
+  sealed trait ProcessStatus
+  case object SHUTDOWN extends ProcessStatus
+  case object STARTING extends ProcessStatus
+  case object STARTED extends ProcessStatus
+  case object SHUTTING_DOWN extends ProcessStatus
```

Comment:
Seems this is only used in BrokerServer, should we move the trait there, or do we intend to use these states in the other server classes as well?

Reply:
> Seems [`sealed trait ProcessStatus`] is only used in BrokerServer, should we move the trait there, or do we intend to use these states in the other server classes as well

We'll use it in `ControllerServer` as well, so I think we should keep it here.